src/main/java/com/powsybl/openloadflow/network/NetworkState.java

```diff
@@ -47,14 +50,16 @@ public static NetworkState save(LfNetwork network) {
         List<BusState> busStates = ElementState.save(network.getBuses(), BusState::save);
         List<BranchState> branchStates = ElementState.save(network.getBranches(), BranchState::save);
         List<HvdcState> hvdcStates = ElementState.save(network.getHvdcs(), HvdcState::save);
-        return new NetworkState(network, busStates, branchStates, hvdcStates, network.getExcludedSlackBuses());
+        List<AreaState> areaStates = ElementState.save(network.getAreaStream().toList(), AreaState::save);
```

Comment:
- in LfNetwork add a new getAreas() method returning the LfNetwork private areas list that you added
- use network.getAreas() here instead of the stream().toList()